WooCommerce/Classes/ViewModels/CardPresentPayments/PaymentCaptureOrchestrator.swift

```diff
@@ -179,6 +181,20 @@ final class PaymentCaptureOrchestrator: PaymentCaptureOrchestrating {
         stores.dispatch(action)
     }
 
+    func presentBackendReceipt(for order: Order, onCompletion: @escaping (Receipt) -> Void) {
+        let action = ReceiptAction.retrieveReceipt(order: order) { result in
+            switch result {
+            case let .success(receipt):
+                onCompletion(receipt)
+            case .failure:
+                // TODO: Error handling. Propagate error.
+                DDLogError("Unable to retrieve receipt for site \(order.siteID) - order \(order.orderID)")
+                break
+            }
+        }
+        stores.dispatch(action)
```

Comment:
Not really from this PR... but it occurs to me that you could consider making this a bit more async/await friendly. See the [WooPaymentsDepositService](https://github.com/woocommerce/woocommerce-ios/blob/trunk/Yosemite/Yosemite/Tools/Payments/WooPaymentsDepositService.swift) for an example of a Yosemite service which does a network call using async await all the way through the chain.

Just a thought, but it would be nice I think?

Reply:
That sound like a good idea. I've logged it here https://github.com/woocommerce/woocommerce-ios/issues/11912 as an enhancement 🙇